lib/Backend/Lower.cpp

```diff
@@ -22435,7 +22435,8 @@ Lowerer::LowerSetConcatStrMultiItem(IR::Instr * instr)
     InsertAdd(false, dstLength, dstLength, srcLength, instr);
 
     dstOpnd->SetOffset(dstOpnd->GetOffset() * sizeof(Js::JavascriptString *) + Js::ConcatStringMulti::GetOffsetOfSlots());
-    this->m_lowererMD.ChangeToAssign(instr);
+
+    LowererMD::ChangeToWriteBarrierAssign(instr);
```

Comment:
Yes, Lei's verification is awesome. Only constraint is that is it depends on TTT repro, and for some we can't catch a TTT.